Zend/tests/load_mangled_object_vars.phpt

```diff
@@ -0,0 +1,84 @@
+--TEST--
+load_mangled_object_vars() function
+--FILE--
+<?php
+
+#[AllowDynamicProperties]
+class A {
+    public $pub = 1;
+    protected $prot = 2;
+    private $priv = 3;
+}
+class B extends A {
+    private $priv = 4;
+}
+
+$obj = new B;
+$obj->dyn = 5;
+$obj->{"6"} = 6;
+
+$vars = get_mangled_object_vars($obj);
+
+$obj2 = new B;
+$obj2->pub = 7;
+load_mangled_object_vars($obj2, $vars);
+
+$vars2 = (array) $obj2;
+var_dump($vars2 === $vars);
+
+#[AllowDynamicProperties]
+class AO extends ArrayObject {
+    private $priv = 1;
+}
+
+$ao = new AO(['x' => 'y']);
+$ao->dyn = 2;
+$vars = get_mangled_object_vars($ao);
+
+$ao2 = new AO();
+load_mangled_object_vars($ao2, $vars);
+
+$vars2 = get_mangled_object_vars($ao2);
+var_dump($vars2 === $vars);
+
+class RO {
+    public function __construct(public readonly int $ro = 123)
+    {
+    }
+}
+
+$ro = new RO;
+load_mangled_object_vars($ro, []);
+var_dump($ro);
+
+load_mangled_object_vars($ro, ['ro' => 'abc']);
```

Comment:
Just here to prove that the implementation is incomplete for now, see PR description.